tracer/src/Datadog.Trace/Propagators/ContextPropagators.cs

```diff
@@ -0,0 +1,78 @@
+// <copyright file="ContextPropagators.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+
+namespace Datadog.Trace.Propagators
+{
+    internal static class ContextPropagators
+    {
+        private static readonly IReadOnlyDictionary<string, Type> AvailablePropagators = new Dictionary<string, Type>
+        {
+            { nameof(Names.Datadog), typeof(DatadogContextPropagator) },
+            { nameof(Names.W3C), typeof(W3CContextPropagator) },
+            { nameof(Names.B3), typeof(B3ContextPropagator) },
+            { nameof(Names.B3SingleHeader), typeof(B3SingleHeaderContextPropagator) },
+            { "B3 single header", typeof(B3SingleHeaderContextPropagator) },
+        };
+
+        public enum Names
+        {
+            Datadog,
+            W3C,
+            B3,
+            B3SingleHeader,
+        }
+
+        public static SpanContextPropagator GetSpanContextPropagator(IEnumerable<string> injectors, IEnumerable<string> extractors)
+        {
+            var propagatorInstances = new Dictionary<string, object>(StringComparer.InvariantCultureIgnoreCase);
+            var lstInjectors = new List<IContextInjector>();
+            var lstExtractors = new List<IContextExtractor>()
```

Comment:
tiny nit: around the variable naming, I'd prefer something like
```suggestion
        public static SpanContextPropagator GetSpanContextPropagator(IEnumerable<string> requestedInjectors, IEnumerable<string> requestedExtractors)
        {
            var propagatorInstances = new Dictionary<string, object>(StringComparer.InvariantCultureIgnoreCase);
            var selectedInjectors = new List<IContextInjector>();
            var selectedExtractors = new List<IContextExtractor>()
```

Also, given we have a max of 4 propagators, is it worth using a list instead of a dictionary, or setting the capacity of the lists?

Reply:
done in [2484784](https://github.com/DataDog/dd-trace-dotnet/pull/2536/commits/2484784f9432b43199e1664a11a052b9820fe52c)